### PR TITLE
[MAINT] transfer tests from pybids

### PR DESCRIPTION
**Pull request by Remi-Gau**
- transfer tests cases that were run in pybids
- split between valid and invalid tests

### CODE REVIEW

**Comment by effigies** [comment]
I'm surprised this is passing.

File: src/bids_validator/test_bids_validator.py [hunk 1]
```diff
@@ -65,9 +65,14 @@ def test_datasets(validator, fname):
...
+        ('/README.rst', True),
         ('/CHANGES', True),
         ('/participants.tsv', True),
+        ('/participants.json', True),
         ('/sub-01/anat/sub-01_T1w.nii.gz', False),
+        ('/RADME', False),  # typo
+        ('/CANGES', False),  # typo
     ],
 )
 def test_top_level(validator, fname, matches):
```
Comment (effigies) [inline src/bids_validator/test_bids_validator.py:78]: In retrospect, I don't like parametrizing on `matches`. I would rather just split it into two functions, e.g.,

```python
@pytest.mark.parametrize('fname', valid_list)
def test_is_top_level(validator, fname):
    assert validator.is_top_level(fname)

@pytest.mark.parametrize('fname', invalid_list)
def test_is_not_top_level(validator, fname):
    assert not validator.is_top_level(fname)
```

If you've got the energy, would you do this? If not, I'll just create an issue, and might sooner or later get around to it.

Comment (Remi-Gau) [reply src/bids_validator/test_bids_validator.py:78]: do you want to apply this to all tests?

Comment (effigies) [reply src/bids_validator/test_bids_validator.py:78]: Yes.

**Comment by effigies** [comment]
@Remi-Gau I assume you're looking through the failing entries, but I can if you'd rather.

**Comment by Remi-Gau** [comment]
yeah I will have a look

**Comment by effigies** [comment]
I suspect that what's happening is that more targeted regexes were expected, and valid BIDS files were being passed to see whether they triggered only the right ones. Checking `is_bids()` would be expected to pass.

**Comment by Remi-Gau** [comment]
OK from my understanding it seems that all the failures are due to the fact that the validaton in pybids used to include methods methods such as `is_anat` which were stricter than the simple `is_bids` here.

I would suggest just removing the test case that fail unless we have any intention of adding those methods back at some point (and even then adding those tests would be easy, so I am very much in favor of removing those test cases).

**Comment by Remi-Gau** [comment]
> I suspect that what's happening is that more targeted regexes were expected, and valid BIDS files were being passed to see whether they triggered only the right ones. Checking `is_bids()` would be expected to pass.

hahaha look at us independently coming to the same conclusion #reproducibility

**Comment by effigies** [comment]
Yeah, no we definitely aren't going to make more and more precise regexes, so let's get rid of these.

I see the regexes as a stopgap until we implement parsing that can tell people what they got wrong, not just "is it BIDS?". Possibly we'll keep them around for fast filtering of non-BIDS files from a layout, as in pybids.

**Comment by Remi-Gau** [comment]
why the failure on 3.8 with minimum dependencies ?

https://github.com/bids-standard/python-validator/actions/runs/10403948653/job/28811412557?pr=8#step:12:439

**Comment by effigies** [comment]
Bad regex generation in earlier bidsschematools. Bump to 0.11 in pyproject.toml and tox.ini.

**Comment by effigies** [comment]
It's pleasing that the `min` check exposed exactly the overlookable edge case it was meant to. I forgot about it, myself.

**Comment by effigies** [comment]
Thanks!

**Pull request merged**